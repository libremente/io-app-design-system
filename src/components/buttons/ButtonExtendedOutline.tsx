import React, { useCallback } from "react";
import {
  GestureResponderEvent,
  Pressable,
  StyleSheet,
  View
} from "react-native";
import Animated, {
  Extrapolate,
  interpolate,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
  withSpring
} from "react-native-reanimated";
import { IOIcons, Icon } from "../icons";
import { H4, LabelSmall } from "../typography";
import { IOScaleValues, IOSpringValues, IOColors, IOStyles } from "../../core";

type Props = {
  label: string;
  description?: string;
  onPress: (event: GestureResponderEvent) => void;
  icon?: IOIcons;
};

const styles = StyleSheet.create({
  button: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingHorizontal: 16,
    paddingVertical: 16,
    borderRadius: 8,
    borderColor: IOColors.bluegreyLight,
    backgroundColor: IOColors.white,
    borderStyle: "solid",
    borderWidth: 1
  }
});

export const ButtonExtendedOutline = ({
  label,
  description,
  onPress,
  icon = "info"
}: Props) => {
  const isPressed: Animated.SharedValue<number> = useSharedValue(0);

  // Scaling transformation applied when the button is pressed
  const animationScaleValue = IOScaleValues?.magnifiedButton?.pressedState;

  const scaleTraversed = useDerivedValue(() =>
    withSpring(isPressed.value, IOSpringValues.button)
  );

  // Interpolate animation values from `isPressed` values
  const animatedStyle = useAnimatedStyle(() => {
    const scale = interpolate(
      scaleTraversed.value,
      [0, 1],
      [1, animationScaleValue],
      Extrapolate.CLAMP
    );

    return {
      transform: [{ scale }]
    };
  });

  const onPressIn = useCallback(() => {
    // eslint-disable-next-line functional/immutable-data
    isPressed.value = 1;
  }, [isPressed]);
  const onPressOut = useCallback(() => {
    // eslint-disable-next-line functional/immutable-data
    isPressed.value = 0;
  }, [isPressed]);

  return (
    <Pressable
      onPress={onPress}
      onPressIn={onPressIn}
      onPressOut={onPressOut}
      accessible={true}
      accessibilityRole={"button"}
    >
      <Animated.View style={[styles.button, animatedStyle]}>
        <View style={IOStyles.flex}>
          <H4>{label}</H4>
          {description && (
            <LabelSmall weight="Regular" color={"bluegreyDark"}>
              {description}
            </LabelSmall>
          )}
        </View>
        <View style={{ marginLeft: 8 }}>
          <Icon name={icon} color="blue" size={24} />
        </View>
      </Animated.View>
    </Pressable>
  );
};

export default ButtonExtendedOutline;
