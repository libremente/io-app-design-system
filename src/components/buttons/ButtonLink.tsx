import React, { useCallback, useMemo } from "react";
import { GestureResponderEvent, Pressable, StyleSheet } from "react-native";
import Animated, {
  Extrapolate,
  interpolate,
  interpolateColor,
  useAnimatedProps,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
  withSpring
} from "react-native-reanimated";
import {
  IOButtonStyles,
  IOColors,
  IOScaleValues,
  IOSpringValues,
  useIOExperimentalDesign
} from "../../core";
import { makeFontStyleObject } from "../../utils/fonts";
import { WithTestID } from "../../utils/types";
import { AnimatedIcon, IOIcons, IconClassComponent } from "../icons/Icon";
import { HSpacer } from "../spacer/Spacer";

type ColorButtonLink = "primary";
export type ButtonLink = WithTestID<{
  color?: ColorButtonLink;
  label: string;
  disabled?: boolean;
  // Icons
  icon?: IOIcons;
  iconPosition?: "start" | "end";
  // Accessibility
  accessibilityLabel?: string;
  accessibilityHint?: string;
  // Events
  onPress: (event: GestureResponderEvent) => void;
}>;

type ColorStates = {
  label: {
    default: string;
    pressed: string;
    disabled: string;
  };
};

const mapColorStates: Record<NonNullable<ButtonLink["color"]>, ColorStates> = {
  // Primary button
  primary: {
    label: {
      default: IOColors["blueIO-500"],
      pressed: IOColors["blueIO-600"],
      disabled: IOColors["grey-700"]
    }
  }
};

// TODO: Remove this when legacy look is deprecated https://pagopa.atlassian.net/browse/IOPLT-153
const mapLegacyColorStates: Record<
  NonNullable<ButtonLink["color"]>,
  ColorStates
> = {
  // Primary button
  primary: {
    label: {
      default: IOColors.blue,
      pressed: IOColors["blue-600"],
      disabled: IOColors["grey-700"]
    }
  }
};

const DISABLED_OPACITY = 0.5;
const IOButtonStylesLocal = StyleSheet.create({
  label: {
    fontSize: 16,
    ...makeFontStyleObject("Regular", false, "ReadexPro")
  }
});

// TODO: Remove this when legacy look is deprecated https://pagopa.atlassian.net/browse/IOPLT-153
const IOButtonLegacyStylesLocal = StyleSheet.create({
  label: {
    fontSize: 16,
    ...makeFontStyleObject("Bold", false, "TitilliumWeb")
  }
});

export const ButtonLink = React.memo(
  ({
    color = "primary",
    label,
    disabled = false,
    icon,
    iconPosition = "start",
    onPress,
    accessibilityLabel,
    accessibilityHint,
    testID
  }: ButtonLink) => {
    const isPressed = useSharedValue(0);
    const { isExperimental } = useIOExperimentalDesign();

    const colorMap = useMemo(
      () => (isExperimental ? mapColorStates : mapLegacyColorStates),
      [isExperimental]
    );
    const buttonStylesLocal = useMemo(
      () => (isExperimental ? IOButtonStylesLocal : IOButtonLegacyStylesLocal),
      [isExperimental]
    );

    // Scaling transformation applied when the button is pressed
    const animationScaleValue = IOScaleValues?.basicButton?.pressedState;

    // Using a spring-based animation for our interpolations
    const progressPressed = useDerivedValue(() =>
      withSpring(isPressed.value, IOSpringValues.button)
    );

    // Interpolate animation values from `isPressed` values
    const pressedAnimationStyle = useAnimatedStyle(() => {
      // Link color states to the pressed states

      // Scale down button slightly when pressed
      const scale = interpolate(
        progressPressed.value,
        [0, 1],
        [1, animationScaleValue],
        Extrapolate.CLAMP
      );

      return {
        transform: [{ scale }]
      };
    });

    const pressedColorLabelAnimationStyle = useAnimatedStyle(() => {
      // Link color states to the pressed states

      const labelColor = interpolateColor(
        progressPressed.value,
        [0, 1],
        [colorMap[color].label.default, colorMap[color].label.pressed]
      );

      return {
        color: labelColor
      };
    });

    // Animate the <Icon> color prop
    const pressedColorIconAnimationStyle = useAnimatedProps(() => {
      const iconColor = interpolateColor(
        progressPressed.value,
        [0, 1],
        [colorMap[color].label.default, colorMap[color].label.pressed]
      );

      return { color: iconColor };
    });

    const AnimatedIconClassComponent =
      Animated.createAnimatedComponent(IconClassComponent);

    const onPressIn = useCallback(() => {
      // eslint-disable-next-line functional/immutable-data
      isPressed.value = 1;
    }, [isPressed]);
    const onPressOut = useCallback(() => {
      // eslint-disable-next-line functional/immutable-data
      isPressed.value = 0;
    }, [isPressed]);

    // Icon size
    const iconSize = 24;

    return (
      <Pressable
        accessibilityLabel={accessibilityLabel}
        accessibilityHint={accessibilityHint}
        accessibilityRole={"button"}
        testID={testID}
        onPress={onPress}
        onPressIn={onPressIn}
        onPressOut={onPressOut}
        onTouchEnd={onPressOut}
        accessible={true}
        disabled={disabled}
        hitSlop={{ top: 14, right: 24, bottom: 14, left: 24 }}
        style={IOButtonStyles.dimensionsDefault}
      >
        <Animated.View
          style={[
            IOButtonStyles.buttonLink,
            iconPosition === "end" && { flexDirection: "row-reverse" },
            disabled ? { opacity: DISABLED_OPACITY } : {},
            /* Prevent Reanimated from overriding background colors
                    if button is disabled */
            !disabled && pressedAnimationStyle
          ]}
        >
          {icon && (
            <>
              {!disabled ? (
                <AnimatedIconClassComponent
                  name={icon}
                  animatedProps={pressedColorIconAnimationStyle}
                  color={colorMap[color]?.label?.default}
                  size={iconSize}
                />
              ) : (
                <AnimatedIcon
                  name={icon}
                  color={colorMap[color]?.label?.disabled}
                  size={iconSize}
                />
              )}
              <HSpacer size={8} />
            </>
          )}
          <Animated.Text
            style={[
              buttonStylesLocal.label,
              disabled
                ? { color: colorMap[color]?.label?.disabled }
                : { color: colorMap[color]?.label?.default },
              !disabled && pressedColorLabelAnimationStyle
            ]}
            numberOfLines={1}
            ellipsizeMode="tail"
            /* A11y-related props:
                DON'T UNCOMMENT THEM */
            /* allowFontScaling
                maxFontSizeMultiplier={1.3} */
          >
            {label}
          </Animated.Text>
        </Animated.View>
      </Pressable>
    );
  }
);

export default ButtonLink;
