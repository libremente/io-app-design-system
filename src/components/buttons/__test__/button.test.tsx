import React from "react";
import { Alert } from "react-native";
import * as TestRenderer from "react-test-renderer";
import ButtonExtendedOutline from "../ButtonExtendedOutline";
import ButtonLink from "../ButtonLink";
import ButtonOutline from "../ButtonOutline";
import ButtonSolid from "../ButtonSolid";

const onButtonPress = () => {
  Alert.alert("Alert", "Action triggered");
};

describe("Test Buttons Components", () => {
  it("ButtonSolid Snapshot", () => {
    const buttonSolid = TestRenderer.create(<ButtonSolid label={"label"} accessibilityLabel={"accessibilityLabel"} onPress={onButtonPress}></ButtonSolid>).toJSON();
    expect(buttonSolid).toMatchSnapshot();
  });

  it("ButtonLink Snapshot", () => {
    const buttonLink = TestRenderer.create(<ButtonLink label={"label"} onPress={onButtonPress} ></ButtonLink>).toJSON();
    expect(buttonLink).toMatchSnapshot();
  });

  it("ButtonOutline Snapshot", () => {
    const buttonOutline = TestRenderer.create(<ButtonOutline label={"label"} accessibilityLabel={"accessibilityLabel"} onPress={onButtonPress} ></ButtonOutline>).toJSON();
    expect(buttonOutline).toMatchSnapshot();
  });

  it("ButtonExtendedOutline Snapshot", () => {
    const buttonExtendedOutline = TestRenderer.create(<ButtonExtendedOutline label={"label"} onPress={onButtonPress}></ButtonExtendedOutline>).toJSON();
    expect(buttonExtendedOutline).toMatchSnapshot();
  });
});
