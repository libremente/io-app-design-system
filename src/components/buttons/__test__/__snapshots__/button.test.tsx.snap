// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Buttons Components ButtonExtendedOutline Snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#CCD4DC",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        },
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        color="black"
        defaultColor="black"
        defaultWeight="SemiBold"
        font="TitilliumWeb"
        fontStyle={
          {
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
        style={
          [
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#0E0F13",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "600",
            },
          ]
        }
        weight="SemiBold"
      >
        label
      </Text>
    </View>
    <View
      style={
        {
          "marginLeft": 8,
        }
      }
    >
      <RNSVGSvgView
        accessibilityElementsHidden={true}
        accessibilityLabel=""
        accessible={false}
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        color={4278219750}
        focusable={false}
        height={24}
        importantForAccessibility="no-hide-descendants"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#0073E6",
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        tintColor={4278219750}
        vbHeight={24}
        vbWidth={24}
        width={24}
      >
        <RNSVGGroup>
          <RNSVGPath
            clipRule={0}
            d="M0 12C0 5.3726 5.3726 0 12 0s12 5.3726 12 12-5.3726 12-12 12S0 18.6274 0 12Zm2 0C2 6.47717 6.47717 2 12 2c5.5228 0 10 4.47717 10 10 0 5.5228-4.4772 10-10 10-5.52283 0-10-4.4772-10-10Zm10.0281-4.50455c.8439 0 1.5281-.68415 1.5281-1.52809 0-.84394-.6842-1.52809-1.5281-1.52809-.844 0-1.5281.68415-1.5281 1.52809 0 .84394.6841 1.52809 1.5281 1.52809Zm0 1.94382h.0005c.5678 0 1.0281.46029 1.0281 1.02813v8.0045c0 .5678-.4603 1.0281-1.0281 1.0281h-.0005C11.4603 19.5 11 19.0397 11 18.4719v-8.0045c0-.56784.4603-1.02813 1.0281-1.02813Z"
            fill={
              [
                2,
              ]
            }
            fillRule={0}
            propList={
              [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`Test Buttons Components ButtonLink Snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 14,
      "left": 24,
      "right": 24,
      "top": 14,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTouchEnd={[Function]}
  style={
    {
      "alignSelf": "flex-start",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "elevation": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "textAlignVertical": "center",
        },
        false,
        {},
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      style={
        [
          {
            "fontFamily": "Titillium Web",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "700",
          },
          {
            "color": "#0073E6",
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`Test Buttons Components ButtonOutline Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "elevation": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "textAlignVertical": "center",
        },
        false,
        {
          "borderWidth": 1,
        },
        {
          "height": 40,
        },
        false,
        {
          "backgroundColor": "rgba(0,115,230,0)",
          "borderColor": "#0073E6",
        },
        {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      ellipsizeMode="tail"
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      style={
        [
          {
            "fontFamily": "Titillium Web",
            "fontStyle": "normal",
            "fontWeight": "700",
          },
          {
            "alignSelf": "center",
          },
          {
            "color": "#0073E6",
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      label
    </Text>
  </View>
</View>
`;

exports[`Test Buttons Components ButtonSolid Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": false,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "elevation": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "textAlignVertical": "center",
        },
        {
          "overflow": "hidden",
        },
        false,
        {
          "height": 40,
        },
        {
          "backgroundColor": "#0073E6",
        },
        {
          "backgroundColor": undefined,
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <View
      entering={[Function]}
      exiting={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        color="white"
        defaultColor="white"
        defaultWeight="Bold"
        ellipsizeMode="tail"
        font="TitilliumWeb"
        fontStyle={
          {
            "fontSize": 16,
          }
        }
        maxFontSizeMultiplier={1.3}
        numberOfLines={1}
        style={
          [
            {
              "alignSelf": "center",
            },
            {
              "fontSize": 16,
            },
            {
              "color": "#FFFFFF",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "700",
            },
          ]
        }
        weight="Bold"
      >
        label
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Buttons Components IconButton Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={8}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      [
        {
          "height": 24,
          "width": 24,
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <RNSVGSvgView
      accessibilityElementsHidden={true}
      accessibilityLabel=""
      accessible={false}
      align="xMidYMid"
      animatedProps={
        {
          "color": undefined,
        }
      }
      bbHeight={24}
      bbWidth={24}
      color={4278219750}
      focusable={false}
      height={24}
      importantForAccessibility="no-hide-descendants"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#0073E6",
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor={4278219750}
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup>
        <RNSVGPath
          clipRule={0}
          d="M12 24c6.629 0 12-5.371 12-12S18.629 0 12 0C5.375 0 0 5.371 0 12s5.375 12 12 12Zm.033-10.97c-.997 0-1.818-.336-2.463-1.014-.646-.674-.968-1.51-.968-2.507 0-1.001.322-1.83.964-2.492.64-.662 1.458-.992 2.458-.992.997 0 1.81.334 2.435 1.017.625.678.94 1.51.94 2.511 0 .993-.315 1.822-.94 2.484-.625.657-1.43.992-2.426.992Zm-3.431 4.905c0-1.001.322-1.83.96-2.492.64-.662 1.458-.992 2.467-.992.996 0 1.809.335 2.43 1.017.625.682.94 1.515.94 2.512l-6.797-.045Z"
          fill={
            [
              2,
            ]
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Test Buttons Components IconButtonContained Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "elevation": 0,
          "justifyContent": "center",
        },
        {
          "borderRadius": 48,
          "height": 48,
          "width": 48,
        },
        {
          "backgroundColor": undefined,
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
      ]
    }
  >
    <RNSVGSvgView
      accessibilityElementsHidden={true}
      accessibilityLabel=""
      accessible={false}
      align="xMidYMid"
      animatedProps={
        {
          "color": undefined,
        }
      }
      bbHeight={24}
      bbWidth={24}
      color={4278219750}
      focusable={false}
      height={24}
      importantForAccessibility="no-hide-descendants"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#0073E6",
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor={4278219750}
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup>
        <RNSVGPath
          clipRule={0}
          d="M12 24c6.629 0 12-5.371 12-12S18.629 0 12 0C5.375 0 0 5.371 0 12s5.375 12 12 12Zm.033-10.97c-.997 0-1.818-.336-2.463-1.014-.646-.674-.968-1.51-.968-2.507 0-1.001.322-1.83.964-2.492.64-.662 1.458-.992 2.458-.992.997 0 1.81.334 2.435 1.017.625.678.94 1.51.94 2.511 0 .993-.315 1.822-.94 2.484-.625.657-1.43.992-2.426.992Zm-3.431 4.905c0-1.001.322-1.83.96-2.492.64-.662 1.458-.992 2.467-.992.996 0 1.809.335 2.43 1.017.625.682.94 1.515.94 2.512l-6.797-.045Z"
          fill={
            [
              2,
            ]
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Test Buttons Components IconButtonSolid Snapshot 1`] = `
<View
  accessibilityLabel="accessibilityLabel"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "flex-start",
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "elevation": 0,
          "justifyContent": "center",
        },
        {
          "borderRadius": 56,
          "height": 56,
          "width": 56,
        },
        {
          "backgroundColor": undefined,
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        {
          "backgroundColor": "#0073E6",
        },
      ]
    }
  >
    <RNSVGSvgView
      accessibilityElementsHidden={true}
      accessibilityLabel=""
      accessible={false}
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      color={4294967295}
      focusable={false}
      height={24}
      importantForAccessibility="no-hide-descendants"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#FFFFFF",
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor={4294967295}
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup>
        <RNSVGPath
          clipRule={0}
          d="M12 24c6.629 0 12-5.371 12-12S18.629 0 12 0C5.375 0 0 5.371 0 12s5.375 12 12 12Zm.033-10.97c-.997 0-1.818-.336-2.463-1.014-.646-.674-.968-1.51-.968-2.507 0-1.001.322-1.83.964-2.492.64-.662 1.458-.992 2.458-.992.997 0 1.81.334 2.435 1.017.625.678.94 1.51.94 2.511 0 .993-.315 1.822-.94 2.484-.625.657-1.43.992-2.426.992Zm-3.431 4.905c0-1.001.322-1.83.96-2.492.64-.662 1.458-.992 2.467-.992.996 0 1.809.335 2.43 1.017.625.682.94 1.515.94 2.512l-6.797-.045Z"
          fill={
            [
              2,
            ]
          }
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
